sync/src/net_time_checker.rs

```diff
@@ -0,0 +1,165 @@
+use ckb_network::{CKBProtocolContext, CKBProtocolHandler, PeerIndex};
+use ckb_protocol::TimeMessage;
+use ckb_util::RwLock;
+use flatbuffers::{get_root, FlatBufferBuilder};
+use log::{debug, warn};
+use std::collections::VecDeque;
+
+const TOLERANT_OFFSET: u64 = 7_200_000;
+const MIN_SAMPLES: usize = 5;
+const MAX_SAMPLES: usize = 11;
+
+/// Collect and check time offset samples
+pub struct NetTimeChecker {
+    /// Local clock should has less offset than this value.
+    tolerant_offset: u64,
+    max_samples: usize,
+    min_samples: usize,
+    samples: VecDeque<i64>,
```

Comment:
check out: https://leetcode.com/problems/find-median-from-data-stream/
solution tab: `Approach #3 Two Heaps! [Accepted]`

Reply:
It's too complex. We only need to maintain 8~1x samples here.

Reply:
the lines of code are similar to the current algorithm, and we can extract it to a util lib and reuse it in `BlockMedianTimeContext`, anyway, I will submit a PR for this improvement

Reply:
Notice below code we also need to maintain last 11 offset samples, so we need a FIFO queue here, it is definitely more complex to implement this in the two-heap algorithm.